.github/workflows/release.yaml

```diff
@@ -0,0 +1,72 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
+  contents: write
+  packages: write
+
+jobs:
+  check:
+    name: preliminary-checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Ensure version is properly set
+        run: |
+          test "$(npm -s run get-version)" = "${{ inputs.tag }}"
+      - name: Give user instruction
+        if: failure()
+        run: echo "::error file=package.json,line=6::The tag ${{ inputs.tag }} must match the version"
+  publish-github:
+    runs-on: ubuntu-latest
+    needs: check
+    steps:
+      - uses: actions/checkout@v4
+      # Setup .npmrc file to publish to GitHub Packages
+      - uses: actions/setup-node@v4
+        with:
+          cache: yarn
+          node-version: '16'
+          registry-url: 'https://npm.pkg.github.com'
+      - run: yarn install --frozen-lockfile
+      - run: yarn publish
```

Comment:
we started to automatically testing workflows (esp. `release` workflows, which have a tendency to break when we need them most...), c.f. [1] : do you know if there is a way to "mock" the npm registry?

the testing framework allows to replace a step (→ do nothing), as a fallback, but it would be nicer to actually validate that publishing is really working (using a mock API or even some sort of local npm registry).

btw, it would probably be good to add similar test in this PR: it is especially useful to ensure the _"preliminary-checks"_ are working as expected, and helps when some overall changes are made (e.g. like we change to workflow so thatversion "input" is sanitized or computed)

[1] https://github.com/scality/vault2/tree/development/8.8/tests/workflows

Reply:
There's way to mock the npm registry yes, some time ago I use to run one locally and wasn't too much trouble to get it up running and start publishing packages to it, you basically run it:
```shell
docker run -it --rm --name verdaccio -p 4873:4873 verdaccio/verdaccio
```
then you configure your package manager to speak with it, with npm it's:
```
npm set registry http://localhost:4873/
```
On the first boot you add a user:
```
npm adduser
```

then you can run a `npm publish` 

Would you see this test running in each repo, or if we happen to share this code through an action or a reusable workflow, having a test like this at the action/workflow location would be enough?